src/app/modules/entry-live/views/live-discovery-chart/live-discovery-request-factory.ts

```diff
@@ -42,35 +43,64 @@ export class LiveDiscoveryRequestFactory implements RequestFactory<KalturaMultiR
     }),
     responseOptions: this._responseOptions
   };
-  
+
   public set dateRange(range: DateRangeServerValue) {
     if (range.hasOwnProperty('toDate') && range.hasOwnProperty('fromDate')) {
       this._dateRange = range;
       this.onPollTickSuccess();
     }
   }
-  
-  constructor(private _entryId: string) {
-    this._getGraphActionArgs.reportInputFilter.entryIdIn = this._entryId;
-    this._getTotalActionArgs.reportInputFilter.entryIdIn = this._entryId;
+
+  constructor(private activationArgs: WidgetsActivationArgs) {
+    this._getGraphActionArgs.reportInputFilter.entryIdIn = this.activationArgs.entryId;
+    this._getTotalActionArgs.reportInputFilter.entryIdIn = this.activationArgs.entryId;
+
+    if (this.activationArgs.countryIn) {
+      this._getGraphActionArgs.reportInputFilter.countryIn = this.activationArgs.countryIn;
+      this._getTotalActionArgs.reportInputFilter.countryIn = this.activationArgs.countryIn;
+    }
+    if (this.activationArgs.regionIn) {
+      this._getGraphActionArgs.reportInputFilter.countryIn = this.activationArgs.regionIn;
+      this._getTotalActionArgs.reportInputFilter.countryIn = this.activationArgs.regionIn;
+    }
+    if (this.activationArgs.citiesIn) {
+      this._getGraphActionArgs.reportInputFilter.countryIn = this.activationArgs.citiesIn;
+      this._getTotalActionArgs.reportInputFilter.countryIn = this.activationArgs.citiesIn;
+    }
+    if (this.activationArgs.deviceIn) {
+      this._getGraphActionArgs.reportInputFilter.countryIn = this.activationArgs.deviceIn;
```

Comment:
@viktor-kyryltsev-kaltura - this should update device and not country

Reply:
@amirch1 changed